filesystem/File.php

```diff
@@ -224,28 +282,27 @@ public static function link_shortcode_handler($arguments, $content = null, $pars
 	/**
 	 * A file only exists if the file_exists() and is in the DB as a record
 	 *
+	 * Use $file->isInDB() to only check for a DB record
+	 * Use $file->File->exists() to only check if the asset exists
+	 *
 	 * @return bool
 	 */
 	public function exists() {
-		return parent::exists() && file_exists($this->getFullPath());
+		return parent::exists() && $this->File->exists();
 	}
 
 	/**
 	 * Find a File object by the given filename.
 	 *
-	 * @param String $filename Matched against the "Name" property.
-	 * @return mixed null if not found, File object of found file
+	 * @param string $filename Filename to search for, including any custom parent directories.
+	 * @return File
 	 */
 	public static function find($filename) {
-		// Get the base file if $filename points to a resampled file
-		$filename = Image::strip_resampled_prefix($filename);
-
 		// Split to folders and the actual filename, and traverse the structure.
 		$parts = explode("/", $filename);
 		$parentID = 0;
 		$item = null;
 		foreach($parts as $part) {
-			if($part == ASSETS_DIR && !$parentID) continue;
```

Comment:
the assets folder shouldn't really be referenced anywhere in the user code anymore. Leaving it in one place would mean having to implement this exception across the board, and would be a risky invitation to users to continue referencing it in their user code.

For example, if we omit the asset when finding an existing filename, we'd have to ensure that creation of new files omitted the asset prefix, otherwise findOrCreate methods could behave unpredictably.

If people try to create a file in the new API with the Filename `assets/parent/file.jpg` then it does less harm to just create the assets parent directory than to try to work around it. As long as the approach is consistent, it shouldn't break anything.